packages/middleware/src/createServer.ts

```diff
@@ -50,11 +52,33 @@ async function createServer(config: MiddlewareConfig): Promise<Express> {
 
   app.post('/:integrationName/:functionName', async (req: Request, res: Response) => {
     const { integrationName, functionName } = req.params as any as RequestParams;
-    const { apiClient, configuration, extensions, customQueries, initConfig } = integrations[integrationName];
+
+    let apiClient;
+    let configuration;
+    let extensions;
+    let customQueries;
+    let initConfig;
+
+    try {
+      if (!integrations[integrationName]) {
+        const errMsg = `"${integrationName}" integration is not configured. Please, check the request path or integration configuration.` +
+          `Request host: ${req?.headers?.host}, origin: ${req.headers.origin}`;
+        throw new Error(errMsg);
+      }
+
+      ({ apiClient, configuration, extensions, customQueries, initConfig } = integrations[integrationName]);
+    } catch (error) {
+      res.status(404);
+      res.send(error);
+
+      return app;
+    }
```

Comment:
```suggestion
    if (!integrations[integrationName]) {
        const errMsg = `"${integrationName}" integration is not configured. Please, check the request path or integration configuration.` +
          `Request host: ${req?.headers?.host}, origin: ${req.headers.origin}`;
        res.status(404);
        res.send(errMsg);
        return;
    }
    
    const { apiClient, configuration, extensions, customQueries, initConfig } = integrations[integrationName];
```

1. What about doing it this way? WDYT?
2. I understand you might want to check also for properties of `integrations[integrationName]` it might be redundant as `registerIntegrations` is supposed to return a certain shape of data anyway. 
3. I wonder why we return `app`, I see we've already had smth like this at the end of the function. Do you know maybe?

I feel like we could also prepare an integration test for this part of the code. It would be smth like [that](https://github.com/vuestorefront/elasticpath-commerce-cloud/pull/271/files) but instead of mocking EPCC, we would mock the package. TBH, I find it interesting and could cover this PR with an integration test in additional PR.

Reply:
1. I did it on purpose, if destructuring will fall because one or more attributes are missing it will throw a verbose error. On the other hand, if you think that we can safely check only the integration then I am ok with the simplified solution.
2. This is basically the same question as 1.
3. IDK, I just did it to align with the rest and keep as small PR as possible (to avoid mistakes and bugs) It is worth checking just not in this PR maybe?
4. I would like to try the integration test on my own if you don't mind. :)

Reply:
1-2. Well... Look at the code of [`registerIntegrations`](https://github.com/vuestorefront/vue-storefront/blob/main/packages/middleware/src/integrations.ts#L71). It should always contain these fields. If a method like `resolveDependency` fails, it will throw an error anyway. If it wouldn't `{ apiClient, configuration, extensions, customQueries, initConfig }` would exist but with nullish value. That's why I think a simplified solution is better.

3. I agree
4. I agree. BTW I think these tests will increase our confidence here a lot. Don't hesitate to hit me for support if you need it. BTW in previous ones I used `nock` for intercepting and mocking HTTP traffic. However, now I would probably go for Mock Service Worker.

Reply:
Ok, I simplified this code.